app/models/volunteer.rb

```diff
@@ -286,6 +287,16 @@ def to_s
 
   private
 
+  def user_added?
+    saved_change_to_attribute?(:user_id)
+  end
+
+  def copy_contact_to_user
+    user.profile&.contact&.update(contact.slice(:first_name, :last_name, :street, :postal_code,
+      :city, :primary_phone, :secondary_phone, :primary_email))
```

Comment:
Is it okay to always overwrite the existing contact, or do we need to preserve existing values?

Reply:
I hit into the problem that exporting users with their `user.profile.contact` data gave me trouble on the volunteer users. Differing names etc., so thats why i did this. Assuming the very most likely case, that a volunteer gets associated with a user only once ever, this is only run once ever.

It was the solution that came up for me, but maybe not perfect. On the other hand it would be also quite miserable to check the `user.role` when exporting the xlsx.

I like neither of the solutions is nice. This is due to the fact that Volunteer has two contact records. One directly and anotherone after Volunteer.user.profile.contact has been created.

Reply:
Okay in that case just leave it as it is now.

Not sure what a clean solution would look like, it would probably make sense to use the contact directly on the volunteer only until the user is created, and then remove the volunteer's contact and only use the user's profile contact from then on.